snuba/replacer.py

```diff
@@ -387,8 +387,15 @@ def process_delete_tag(message, dataset) -> Optional[Replacement]:
 
     assert isinstance(tag, str)
     timestamp = datetime.strptime(message["datetime"], settings.PAYLOAD_DATETIME_FORMAT)
-    tag_column_name = dataset.get_tag_column_map()["tags"].get(tag, tag)
-    is_promoted = tag in dataset.get_promoted_tags()["tags"]
+    promoted_col_spec = (
+        dataset.get_dataset_schemas()
+        .get_read_schema()
+        .get_data_source()
+        .get_promoted_columns_spec()
```

Comment:
At a point we may want to simplify the way to access the data source given a dataset.